ethcore/src/client/client.rs

```diff
@@ -2160,18 +2160,16 @@ impl IoClient for Client {
 		let hash = unverified.hash();
 		{
 			// check block order
-			if self.chain.read().is_known(&hash) {
+			let parent_hash = unverified.parent_hash();
+			let status = self.block_status(BlockId::Hash(parent_hash));
```

Comment:
Fetching the status of `parent_hash` now instead of `hash`

Reply:
Yeah you're right. I think I got this PR wrong.